gslib/commands/rm.py

```diff
@@ -393,10 +393,15 @@ def RemoveFunc(self, name_expansion_result, thread_state=None):
 
     exp_src_url = name_expansion_result.expanded_storage_url
     self.logger.info('Removing %s...', exp_src_url)
-    gsutil_api.DeleteObject(exp_src_url.bucket_name,
-                            exp_src_url.object_name,
-                            preconditions=self.preconditions,
-                            generation=exp_src_url.generation,
-                            provider=exp_src_url.scheme)
+    try:
+      gsutil_api.DeleteObject(exp_src_url.bucket_name,
+                              exp_src_url.object_name,
+                              preconditions=self.preconditions,
+                              generation=exp_src_url.generation,
+                              provider=exp_src_url.scheme)
+    except NotFoundException as e:
+      # Retries during recursive object deletes can cause harmless 404s.
+      self.logger.info('Cannot find %s', exp_src_url)
```

Comment:
Curious about why we want to retry on 404. Can you please add a link to the issue this is trying to fix?

Reply:
b/148433922